projboard/tests/test_article.py

```diff
@@ -0,0 +1,142 @@
+        article = create_articles[0]
+        # testing @property functions => PartOf..()
+        # whent title & content length is less than reality
+        assert article.PartOfTitle(5) == article.title[:5] + '...'
+        assert article.PartOfContent(5) == article.content[:5] + '...'
+        # whent title & content length is greater than reality
+        assert article.PartOfTitle(100) == article.title[:100]
+        assert article.PartOfContent(100) == article.content[:100]
+
+    # @pytest.mark.skip
+    @pytest.mark.django_db
+    def test_objectSavedInDb(self, create_articles):
+        assert create_articles[0] in Article.objects.all()
+        assert create_articles[1] in Article.objects.all()
+
+    # @pytest.mark.skip
+    @pytest.mark.django_db
+    # using Edite function
+    def test_EditObject(self, create_articles, create_subjects):
+
+        article = create_articles[0]
+        title = "New Title"
+        content = "New Content"
+        subject = create_subjects[1]
+        old_subject = create_subjects[0]
+
+        assert article.subject_id == old_subject
+
+        article.Edit(title, content, subject)
+
+        assert article.title == title
+        assert article.content == content
+        assert article.subject_id == subject
+        # if subject not in Article.SUBJECTS the value won't be changed
+        unsaved_subject = Subject(name='k')
+        article.Edit(subject=unsaved_subject)
+        assert article.subject_id == subject
+
+    # @pytest.mark.skip
+    @pytest.mark.django_db
+    def test_searchObjects(self, create_articles, create_subjects, create_user):
+        user1 = create_user[1]
+        article_list = [i for i in create_articles]
+        subject1 = create_subjects[1]
+
+        # search by title or a part of it
+        assert article_list[1] in Article.searchByTitle("Test")
+        assert article_list[1] not in Article.searchByTitle("Hakuna matata")
+
+        # search by user
+        assert article_list[1] in Article.searchByUser(user1)
+        assert article_list[0] not in Article.searchByUser(user1)
+
+        # search by subject
+        assert article_list[1] in Article.searchBySubject(subject1)
+        assert article_list[0] not in Article.searchBySubject(subject1)
+
+    @pytest.fixture
+    @pytest.mark.django_db
+    def create_like(self, create_user, create_articles):
+        like = Like(user_id=create_user[0], article_id=create_articles[0])
+        like.save()
+        return like
+
+    @pytest.fixture
+    @pytest.mark.django_db
+    def create_view(self, create_user, create_articles):
+        view = View(user_id=create_user[0], article_id=create_articles[0])
+        view.save()
+        return view
+
+    # @pytest.mark.skip
+    @pytest.mark.django_db
+    def test_filterObjects(self):
+        # Filter By Likes
+        print(Article.filterByLikes())
+        # Filter By Views
+        print(Article.filterByViews())
+        assert True
```

Comment:
What are you asserting in this test? 
It will never fail,

Reply:
filterBy fuctions returns a list of all articles ordered by num of like from greatest to least , Im printing the list .

Reply:
in order to claculate these functions i should do a query, but if so its gonna be the same query that in the method it self , so it's gonna look like => filterByViews == filterByViews

Reply:
@Yarboa

Reply:
if it is important, please assert that they are odered